### PR TITLE
PR landing page from results page

### PR DESCRIPTION
**Pull request by bridgettorres**
Created Results Page with minor changes to Landing Page. Added tests for landing and results page.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,7 +13,6 @@
 .pub/
 build/
 coverage/
-lib/generated_plugin_registrant.dart
```
Comment (erickjtorres) [inline .gitignore]: why did you remove this?

File: lib/results_page.dart [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+    companyName: "Cool Air HVAC",
+    ownerName: "Sarah Brown",
+    image: "/placeholder.svg?height=100&width=100",
+    rating: 4.7,
+    tags: ["HVAC", "Air Conditioning", "Heating"],
+  ),
+];
+
+
+class ResultsPage extends StatelessWidget{  //defined class that extends statefulwidget (means app changes over time)
```
Comment (erickjtorres) [inline lib/results_page.dart]: Too much information in the comments. No need for all of this. please remove

File: lib/results_page.dart [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+         ],)
+      ],
+    ),
+  );
+}
+}
+
+//creating a class for the tags
+//this can be added later: this widget accepts text and color, in which depending on the tag, the color of the oval should be different
+class OvalTags extends StatelessWidget {
```
Comment (erickjtorres) [inline lib/results_page.dart]: For any public classes is preferred to keep them in seperate files. Please pull this out into its own

File: lib/results_page.dart [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+              fontSize: 12.0,
+          )
+
+        ))
+    );
+  }
+}
+
+/*
+    Contractor(
```
Comment (erickjtorres) [inline lib/results_page.dart]: Remove

File: lib/results_page.dart [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+    ),
+*/
+
+class ProfileCard extends StatelessWidget {
+  final int id;
+  final String companyName;
+  final String ownerName;
+  final String image;
+  final double rating;
+  //final List tags;
```
Comment (erickjtorres) [inline lib/results_page.dart]: nit remove comment

File: pubspec.yaml [hunk 6]
```diff
@@ -35,6 +35,7 @@ dependencies:
   # The following adds the Cupertino Icons font to your application.
   # Use with the CupertinoIcons class for iOS style icons.
   cupertino_icons: ^1.0.8
+  mockito: ^5.0.0
```
Comment (erickjtorres) [inline pubspec.yaml]: let us use https://pub.dev/packages/mocktail instead. Please update the tests with Mocktail

File: test/landing_page_test.dart [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+      //expect(find.text('filter'), findsNothing);
+      final ebfinder = find.byType(ElevatedButton);
+      await tester.tap(ebfinder);
+      await tester.pumpAndSettle(); //wait for navigation to complete
+
+      expect(find.text('Find who you need'), findsNothing);
+
+  });
+
+    // testWidgets('Button is present and triggers navigation after tapped', (WidgetTester tester) async {
```
Comment (erickjtorres) [inline test/landing_page_test.dart]: nit remove commented code

File: test/landing_page_test.dart [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+      home: MyHomePage(title: '',),
+      )
+    );
+    expect(find.text('Find who you need'), findsOneWidget);
+  });
+  testWidgets('Button is present and triggers navigationg after tapped', (tester)async{
+    await tester.pumpWidget(const MaterialApp(
+      home: MyHomePage(title: '',),),);
+      expect(find.text('Find who you need'), findsOneWidget);
+      //expect(find.text('filter'), findsNothing);
```
Comment (erickjtorres) [inline test/landing_page_test.dart]: remove

**Review by erickjtorres — CHANGES_REQUESTED** [review]
Same thing as previous PR. But a couple more notes:

1. Keep public classes/widgets in seperate files
2. Let us use mocktail instead of mockito
3. Same thing as before remove any unnecessary comments

File: pubspec.yaml (hunk 6 above)
Comment (bridgettorres) [reply pubspec.yaml]: Mockito was originally what we were gonna use but we did not end up using it for our test cases. Do you still want me to add tests that use Mocktail. I also added Mocktail to dependencies.

File: lib/results_page.dart (hunk 3 above)
Comment (bridgettorres) [reply lib/results_page.dart]: Just to clarify did you also want me to put ProfileCard in a separate file outside of results page?

Comment (erickjtorres) [reply lib/results_page.dart]: Every public class should be in a seperate file (in other words every stateless widget). If you think the classes won't be reused you can make them private and keep them in the same file

File: pubspec.yaml (hunk 6 above)
Comment (erickjtorres) [reply pubspec.yaml]: Yes update to mocktail

File: lib/main.dart [hunk 9]
```diff
@@ -13,10 +14,11 @@ class MyApp extends StatelessWidget {
     return MaterialApp(
       title: 'Flutter Demo',
       theme: ThemeData(
+        scaffoldBackgroundColor: const Color.fromARGB(255, 77, 117, 79),
 
         useMaterial3: true,
       ),
-      home: const MyHomePage(title: 'Flutter Demo Home Page'),
+      home: const ResultsPage(),
```
Comment (inikasingh21) [inline lib/main.dart]: are we leaving it like this for testing purposes?

**Review by inikasingh21 — DISMISSED** [review]

File: lib/main.dart (hunk 9 above)
Comment (erickjtorres) [reply lib/main.dart]: yeah can we change this back?

**Review by erickjtorres — DISMISSED** [review]

**Review by erickjtorres — APPROVED** [review]

**Review by inikasingh21 — APPROVED** [review]
